### PR TITLE
Appointment request model

### LINKED ISSUE #29: Design DB structure & create model for mentees' appointment requests
#### Description
Mentees will fill out a form on the frontend to request an appointment with a specific mentor. This will be for a specific time and include the following information:
image

As per the [PRD](https://docs.google.com/document/d/14DHPbMC_1VVO7HG_0sKgJ_gMHOcJxmWx7u0Q4unzbN0/edit).

This will then create a request to the mentor. The different states of a request can be pending, accepted, denied. 

#### Tasks
- [ ] Create DB structure
- [ ] Present to team, get feedback until we get a final structure
- [ ] Implement the model

### PR DESCRIPTION
**Pull request by nroy20**
## Description :star2:

Implementation of model for all requests for mentor appointments with the following requirements:
image

Fixes #29
 
## TODOs :star:
- [x] Design model
- [x] Implement model
- [ ] Review

### CODE REVIEW

File: backend/api/models/ApptRequest.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from .base import db
+from mongoengine import *
+from flask_mongoengine import Document
+from api.models import Email
+
+class ApptRequest(Document, Mixin):
+    """Appointment Request Collection."""
+
+    name = StringField(required = True)
+    email = EmbeddedDocumentField(Email, required = True)
```
Comment (chyku) [inline backend/api/models/ApptRequest.py:11]: don't use the email document LOL it was only an example and should be fine as a string

File: backend/api/models/ApptRequest.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from api.core import Mixin
+from .base import db
+from mongoengine import *
+from flask_mongoengine import Document
+from api.models import Email
+
+class ApptRequest(Document, Mixin):
```
Comment (chyku) [inline backend/api/models/ApptRequest.py:7]: hmm idk if it's necessary to abbreviate here

File: backend/api/models/ApptRequest.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    ethnicity = StringField(required = True)
+    location = StringField()
+    mentorship_goals = StringField(required = True)
+    specialist_categories = ListField(StringField(), required = True)
+    message = StringField()
+    attendee_count = IntField(required = True)
+    organization = StringField(required = True)
+
+    def __repr__(self):
+        return f"""<name: {self.name}
```
Comment (chyku) [inline backend/api/models/ApptRequest.py:25]: include the name of the model here

**Review by chyku — CHANGES_REQUESTED** [review]

**Review by Leundai — CHANGES_REQUESTED** [review]
Make sure to add your model to ```__init__.py``` 👍 
image

File: backend/api/models/AppointmentRequest.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+from api.core import Mixin
+from .base import db
+from mongoengine import *
+from flask_mongoengine import Document
+from api.models import Email
```
Comment (chyku) [inline backend/api/models/AppointmentRequest.py:5]: unnecessary import now

File: backend/api/models/AppointmentRequest.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class AppointmentRequest(Document, Mixin):
+    """Appointment Request Collection."""
+
+    name = StringField(required=True)
+    email = StringField(required=True)
+    phone_number = StringField(required=True)
+    languages = ListField(StringField(), required=True)
+    age = IntField(required=True)
```
Comment (chyku) [inline backend/api/models/AppointmentRequest.py:15]: now that we have age ranges, we can make this a string

**Review by chyku — CHANGES_REQUESTED** [review]
and also export in __init__.py

File: backend/api/models/AppointmentRequest.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    ethnicity = StringField(required=True)
+    location = StringField()
+    mentorship_goals = StringField(required=True)
+    specialist_categories = ListField(StringField(), required=True)
+    message = StringField()
+    attendee_count = IntField(required=True)
+    organization = StringField(required=True)
+
+    def __repr__(self):
+        return f"""<AppointmentRequest name: {self.name}
```
Comment (chyku) [inline backend/api/models/AppointmentRequest.py:29]: make sure to update these with the fields added, at least mentorid

File: backend/api/models/__init__.py [hunk 7]
```diff
@@ -8,6 +8,7 @@
 from .Video import Video
 from .Availability import Availability
 from .MentorProfile import MentorProfile
+from .AppointmentRequest import AppointmentRequest
```
Comment (chyku) [inline backend/api/models/__init__.py:11]: not sure if you deleted the export, but you need to add appointmentrequest to `__all__`

**Review by chyku — CHANGES_REQUESTED** [review]
small things, otherwise looks good

**Review by chyku — APPROVED** [review]
lgtm

**Review by Leundai — APPROVED** [review]
Looks good

**Pull request merged**